File: Backend/Blog.Core/Entities/Post.cs
namespace Blog.Core.Entities;

public class Post
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public required string MarkdownContent { get; set; }
    public DateTime Created { get; set; }
    public Guid AuthorId { get; set; }
    public User? Author { get; set; }
}

File: Frontend/src/entities/model/post.ts
import type {Guid} from "guid-typescript";
import type {User} from "./user.ts";

export type Post = {
    postId: Guid;
    name: string;
    description: string;
    markdownContent: string;
    created: Date;
    authorId: Guid;
    author?: User;
}